### PR TITLE
XM-CL ANCW: Caustic Armor for research

### PR DESCRIPTION
**Pull request by TheManWithNoHands**
# About the pull request

Adds a new set of armor to the game. 
The XM-CL ANCW. An Armor set that research can print for a step price.
It provides heavy Bio protection, a bit better bomb protection and heat protection, but in turn, it is much less effective against bullets and reacts harsher to electric shocks.

It has even an actual place in the lore, being that its based on the Z-110 ANCW. An anti xenomprh armor that uses gel to turn acid into basicly water. The chest pice contains its lore blurp.
https://avp.fandom.com/wiki/Z-110_Acid_Neutralizing_Combat_Wardrobe

Prices are WIP and definitly need a second opion.

# Explain why it's good for the game

Its another thing research can buy using their printer. 
The bioprinter is currently our fairest pointshop we have, being the only one that doesnt gain points by being AFK. And we have one shop where you can be AFK to earn metal, wich is one of the strongest things in the game.

The printer already has a lot of great options people can chose to spend their points on. Expanding the list of avalibel things to buy from it, gives the printer and research much more varity. So they hopefully dont do the same every round.
There is already the thing that a lot of researchers only use the printer to make more points for stims.
So giving the printer more atractive options would help reduce the over reliance on stims.

# Testing Photographs and Procedure

https://imgur.com/a/TpXl0oc

# Changelog
:cl: TheManWithNoHands

add: Added Caustic Armor
imageadd: added icons for new armor, helmet, gloves, boots, uniform and kit

/:cl:

### CODE REVIEW

**Comment by TheManWithNoHands** [comment]
Always make sure not to slip on the enter key kids

**Comment by TheManWithNoHands** [comment]
Always make sure not to slip on the enter key kids.

File: code/datums/research_upgrade_datum.dm
Comment (Zonespace27) [inline code/datums/research_upgrade_datum.dm:225]:
```suggestion
		/obj/item/clothing/under/marine/caustic,
    )
```

File: code/modules/clothing/head/helmet.dm
Comment (Zonespace27) [inline code/modules/clothing/head/helmet.dm:234]: Inherited vars above new ones

File: code/modules/clothing/head/helmet.dm
Comment (Zonespace27) [inline code/modules/clothing/head/helmet.dm:202]: This should be a marine helmet subtype

File: code/modules/clothing/head/helmet.dm
Comment (Zonespace27) [inline code/modules/clothing/head/helmet.dm:230]: Specifically to avoid this var junk that shouldn't work, even if it does

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/helmet.dm
Comment (TheManWithNoHands) [reply code/modules/clothing/head/helmet.dm:202]: I was scared of all the overlays that came with it. But ok. Easy fix.

File: code/datums/research_upgrade_datum.dm
Comment (TheManWithNoHands) [reply code/datums/research_upgrade_datum.dm:225]: fucking hate how my keybord lags

File: code/datums/research_upgrade_datum.dm
Comment (kiVts) [inline code/datums/research_upgrade_datum.dm]: did you try and see if this works? I have a high suspicion it will spawn first thing in the list if it will spawn at all.

Comment (kiVts) [reply code/datums/research_upgrade_datum.dm]: I recommend adding a kit (e.g. gun kit, or spec kit) so item reference has one path

Comment (kiVts) [reply code/datums/research_upgrade_datum.dm]: or code it so it accepts lists, shouldn't be too hard either

Comment (TheManWithNoHands) [reply code/datums/research_upgrade_datum.dm]: doesnt spawn anythingg. But i had to decide between learning how to fix this or finally sleeping. Yeah, sleep won. 
The error message the printer gives is from it having an incorect path.

Comment (kiVts) [reply code/datums/research_upgrade_datum.dm]: Yeah, because the path is literally a list.

Comment (TheManWithNoHands) [reply code/datums/research_upgrade_datum.dm]: Hey, to be fair, i know coding by copy pasting what i saw works in other places. Not by being smart.
Like how i realise rigth now i probably should look at how vendors put out packs of stuff.

Comment (kiVts) [reply code/datums/research_upgrade_datum.dm]: we can talk about how to go about that in discord

File: code/modules/clothing/head/helmet.dm
Comment (Zonespace27) [inline code/modules/clothing/head/helmet.dm:1018]: You don't seem to ever use these?

File: code/modules/clothing/suits/marine_armor/_marine_armor.dm
Comment (Zonespace27) [inline code/modules/clothing/suits/marine_armor/_marine_armor.dm:674]:
```suggestion
	desc = "A lightweight piece of Z-210 ANCW armor. It has incredibly soft cushioning on the inside, thanks to the gel pads. A truck could hit you and you be fine in this."
```

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/modules/clothing/head/helmet.dm
Comment (TheManWithNoHands) [reply code/modules/clothing/head/helmet.dm:1018]: Oh rigth, went a bit braindead for a moment there. Copyd that from our gasmasks. Let me look for the code to make this do something.

Ok, im back. And fun thing, thats not the only thing that doesnt do anything.
`var/gas_transfer_coefficient`
`var/permeability_coefficient`
They are both on a ton of headgear and masks, but i could not find them even mentioned other then as an item var.
Non of those two things seem to do anything. They are just vars that seem to be left over.

Comment (TheManWithNoHands) [reply code/modules/clothing/head/helmet.dm]: removed slop in armor

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by Catzonwheelz** [comment]
More things for biopoints is good. but how does it fair up with boiler nero gas?

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by TheManWithNoHands** [comment]
Doesnt do shit against any kind of gas. There are no gas filters after all, the ones we have are make belive.
Only effects spit acid, prea acid ball and acid spray.
Not sure about body acid though.

The one thing this armor does is allowing you to flip a spitter off because they are useless against you.

**Comment by TheManWithNoHands** [comment]
Not gonna resolve merge conflicts until this is ready for merge. MY pc is fuckd so i cant install the hooks and need to resolve it manually.
Aint doing that every time.

**Comment by TheManWithNoHands** [comment]
Thats very helpful, thank you. I posted about this in the wiki chanel on discord already but got no feedback there sadly.
gonna work on all that to properly implement it.

Gonna drop the Grand copr refrence and just keep it as the material i mentioned in the lore blurp, xeno spit. Only said it offhand in the lore blup on purpose. So you reading "oh this is an interesting pice of science" and you sligthly skim over the word "spit" and take a minute to realise what that means.
Im pretty sure you missed that detail because of that.

I found a couple refrences that xenos insides are still somewhat vunrebal to acids. So it would make sense if their own saliva can neutralise acid that is on the victim they swallow. That was the reason why it needs biomatter points to produce it. Becauce without the compound, its literally like wearing a kiddy pool around your chest with 10$ storm trooper armor.

But on some points:
* Good point on the renaming of the armor. But i will need the whole set to be the same name. Also i dont think calling it ANCW outside its descpritions is a good idea. Name needs to be short and snappy after all. Likly just gonna call it XM3 if im lazy.
* Not gonna make it actually use up rescources. Thats mostly just flavor to use already exsiting lore for it. Gameplay first, lore second. Having to actually refill it would be a pain in the ass after all, and nobody will pick it. The Z-100 did magicly make infinit ammounts of its gell, so i think having the stored gel be a compount that expands drasticly into a gel under the rigth condicions is fine. Basicly like spray foam. Better then just making gel from nothing.
* I did plan on giving the Gas mask a use, but i then found out we have no system at all in place for gas masks to use. Every masks has some var for that stuff, but they are all empty. Every gasmask we have is purly cosmetical. And i dont know how to make a gas filter system from scratch. Besides, this way the armor isnt dummy OP and plays more fair against. Imagine if the armor made marines able to tank spitters and also boilers. Xeno mains would riot.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Review by nauticall — APPROVED** [review]
orange in the end, huh? looks good too, sprites 👍

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Review by hry-gh — CHANGES_REQUESTED** [review]
fix ci failure. you can press details to read what failed

**Review by kiVts — CHANGES_REQUESTED** [review]
gramar

File: code/modules/clothing/suits/marine_armor/_marine_armor.dm
Comment (kiVts) [inline code/modules/clothing/suits/marine_armor/_marine_armor.dm:698]: on an unrelated note, are you sure this is necessary?

Comment (TheManWithNoHands) [reply code/modules/clothing/suits/marine_armor/_marine_armor.dm:698]: Yes, it needs the NO_NAME_OVERRIDE and NO_SNOW_TYPE thing. Otherwise it will pull a non exsitent sprite if its on a snow map. Says darek.

But i do realise i dont need to define some stuf twice. I will remove the icon defines and then check.

Comment (kiVts) [reply code/modules/clothing/suits/marine_armor/_marine_armor.dm:698]: I mean the parent_type::flags_atom

Comment (kiVts) [reply code/modules/clothing/suits/marine_armor/_marine_armor.dm]: There are reasons why its not used. Mostly because it has the potential to be hacky which is bad and thats bad and so on. I dont recommend to use it tbh. This should be fine though, probably.

Comment (kiVts) [reply code/modules/clothing/suits/marine_armor/_marine_armor.dm]: nvm its new thats why its not used lol

Comment (TheManWithNoHands) [reply code/modules/clothing/suits/marine_armor/_marine_armor.dm]: > There are reasons why its not used. Mostly because it has the potential to be hacky which is bad and thats bad and so on. I dont recommend to use it tbh. This should be fine though, probably.

Even i know not to asume things that hard.
Also its used like 4 other times.

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by TheManWithNoHands** [comment]
im not gonna touch the .dmi conflicts until the pr is aproved to be merged balance wise. I have to do this by hand everytime a Pr gets merged that touches any of those files. NO.

File: code/datums/research_upgrade_datum.dm
Comment (kiVts) [inline code/datums/research_upgrade_datum.dm:214]:
```suggestion
	desc = "Experimental armor set using the ANCW system to withstand acid."
```

File: code/modules/cm_marines/equipment/kit_boxes.dm
Comment (kiVts) [inline code/modules/cm_marines/equipment/kit_boxes.dm:586]: add a description

File: code/modules/clothing/shoes/marine_shoes.dm
Comment (kiVts) [inline code/modules/clothing/shoes/marine_shoes.dm:85]:
```suggestion
	desc = "A pair of ANCW Boots. Their insides are cushioned with soft gel pads to give you an optimal footbed. You wish you had a pair of these at home."
```

File: code/modules/clothing/head/helmet.dm
Comment (kiVts) [inline code/modules/clothing/head/helmet.dm:1001]:
```suggestion
	desc = "It's an ANCW Helmet. It is fitted with an integrated gas mask and visor. The inside is cushioned with cool gel pads to protect against harsh temperature changes and heavy impacts."
```

File: code/datums/research_upgrade_datum.dm
Comment (kiVts) [inline code/datums/research_upgrade_datum.dm:213]:
```suggestion
	name = "XM-CL Armor Kit"
```

**Review by kiVts — CHANGES_REQUESTED** [review]
fix conflicts, The price might be too high for what it is but we will see.

File: code/datums/research_upgrade_datum.dm
Comment (TheManWithNoHands) [reply code/datums/research_upgrade_datum.dm:214]: It is a part of an armor set you goober

File: code/modules/cm_marines/equipment/kit_boxes.dm
Comment (TheManWithNoHands) [reply code/modules/cm_marines/equipment/kit_boxes.dm:586]: its a kit, they dont need one.

File: code/modules/clothing/shoes/marine_shoes.dm
Comment (TheManWithNoHands) [reply code/modules/clothing/shoes/marine_shoes.dm:85]: Words chosen deliberatly.

**Comment by TheManWithNoHands** [comment]
> fix conflicts, The price might be too high for what it is but we will see.

I keep having to resovle the conflix again and again by hand because i cant make the hooks work. So no, im waiting until this is aproved balance wise. Also stop setting my Pr on draft, especially if half your suggestions are just about subjective phrasing.

File: code/datums/research_upgrade_datum.dm
Comment (kiVts) [reply code/datums/research_upgrade_datum.dm:214]: you get the whole kit though?

**Comment by Drulikar** [comment]
> > fix conflicts, The price might be too high for what it is but we will see.
> 
> I keep having to resovle the conflix again and again by hand because i cant make the hooks work. So no, im waiting until this is aproved balance wise. The reason why this isnt going stale is because i used to immediatly .dmi errors.
> 
> Also stop setting my Pr on draft, especially if half your suggestions are just about subjective phrasing.

You should seek help getting the hooks working. As far as I knew they were working for you. The last thing you were told to do was to clear your python cache so you can install the new python version.

It is standard procedure for maintainers to draft a PR when a review is made. A drafted PR & conflicted PR is remove from the review queue.

**Comment by TheManWithNoHands** [comment]
> You should seek help getting the hooks working. As far as I knew they were working for you. The last thing you were told to do IIRC was to clear your python cache so you can install the new python version.

Yeah, i will tinker around a bit again and see if i can get it to work. If not, i ask on the discord again.

> It is standard procedure for maintainers to draft a PR when a review is made. A drafted PR & conflicted PR is remove from the review queue.
Ahh, got very confused by it. First time i saw it happen like that. Sorry about that then.

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

**Comment by TheManWithNoHands** [comment]
definitly needs some buffs

**Comment by kiVts** [comment]
Off testmerge because it conflicts with hybrisa now

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

**Comment by TheManWithNoHands** [comment]
something something, needs likly a rework and/or major rebalancing

**Comment by harryob** [comment]
pr appears to be stale (no commits for a couple weeks), feel free to ask us to reopen/open a new pr when you're working on it again

**Pull request closed without merging**
